### DISCUSSION TITLE
How do I get user agent or any header from a request inside a middleware?

### DISCUSSION THREAD

**Discussion by Extarys** [nuxt/nuxt#17373] [Questions] [answered]
How do I get the user agent in a a server middleware? 😂
I searched https://www.jsdocs.io/package/h3 in length but didn't see any header shortcut, the source code wasn't that helpful to me either.

When I print the `req` object I see `Symbol(kHeaders)` but no easy way to access it.

`server/middleware/headerAuth.ts`:
```
export default defineEventHandler((event) => {
// Extract user-agent
})
```

Thank you!

**Comment by Diizzayy** [comment, marked as answer]
@Extarys You can access the user agent headers as seen below

`nuxt-app/server/middleware/ua.ts`
```ts
export default defineEventHandler((event) => {
    const agent = event?.req?.headers?.['user-agent']
    console.log('agent', agent);
})
```

**Reply by Extarys** [reply]
No idea how I didn't see it in the object, thank you!
